core/src/main/java/io/camunda/migrator/interceptor/VariableInvocation.java

```diff
@@ -14,40 +14,45 @@
  */
 public class VariableInvocation {
 
-  protected VariableInstanceEntity variable;
-  protected String processInstanceId;
-  protected String activityInstanceId;
+  protected VariableInstanceEntity c7Variable;
+  protected MigrationVariableDto migrationVariable;
 
   public VariableInvocation(VariableInstanceEntity variable) {
-    this.variable = variable;
-    this.processInstanceId = variable.getProcessInstanceId();
-    this.activityInstanceId = variable.getActivityInstanceId();
+    if (variable == null) {
+      throw new IllegalArgumentException("Variable cannot be null");
+    }
+    this.c7Variable = variable;
+    this.migrationVariable = new MigrationVariableDto(variable.getName(), variable.getValue());
   }
 
   /**
    * Returns the variable instance entity associated with this invocation.
    *
    * @return the {@link VariableInstanceEntity}
    */
-  public VariableInstanceEntity getVariable() {
-    return variable;
+  public VariableInstanceEntity getC7Variable() {
+    return c7Variable;
   }
 
   /**
-   * Returns the process instance ID associated with this variable invocation.
+   * Returns the DTO representation of the variable.
    *
-   * @return the process instance ID
+   * @return the {@link MigrationVariableDto}
    */
-  public String getProcessInstanceId() {
-    return processInstanceId;
+  public MigrationVariableDto getMigrationVariable() {
+    if (migrationVariable == null) {
```

Comment:
🙃  Is it possible for `migrationVariable` to be null? There is only 1 constructor and that sets `migrationVariable`